// Copyright 2022 ByteDance and/or its affiliates.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax="proto2";

package monolith.model_export;

message ServingEntry {
  optional string export_dir = 1;
  
  // Records the global step for exported model.
  optional int64 global_step = 2;

  // TODO(leqi.zou): Add deps to support better recovery.
}

message ServingModelState {
  repeated ServingEntry entries = 1;
}

